handler/user_handler.go

```diff
@@ -60,3 +59,37 @@ func LoginHandler(userService service.UserService) http.HandlerFunc {
 		w.Write(responseBytes)
 	}
 }
+
+func ListUsersHandler(userService service.UserService) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		w.Header().Set("Content-Type", "application/json")
+
+		user, err := userService.ListUsers(r.Context())
+
+		if err == customerrors.NoUsersExist {
+			fmt.Println("handler: No users exist")
+			w.WriteHeader(http.StatusNotFound)
+			responseBytes, _ := json.Marshal(contract.ErrorResponse{Error: "no user found"})
+			w.Write(responseBytes)
```

Comment:
in case of list sort of API, the api response should not give error code in response if no entries are found.
You can return empty array in the response with status code 200 OK

Reply:
@vkhichar any thoughts?

Reply:
agreed. though, merging this MR, will take that as an enhancement